src/lib/raft/raft.c

```diff
@@ -367,6 +408,30 @@ raft_process_recovery(struct raft *raft, const struct raft_msg *req)
 	assert(!raft_is_enabled(raft));
 }
 
+/** Update raft state once leader is seen. */
+static bool
+raft_leader_see(struct raft *raft);
+
+/**
+ * Clear leader-related fields on leader loss or resign and let everyone know
+ * this node doesn't see the leader anymore.
+ */
+static inline void
+raft_leader_resign(struct raft *raft)
+{
+	assert(raft->leader == raft->self);
+	/*
+	 * Update leader_last_seen when resigning so that leader_idle
+	 * starts counting from zero after the resign.
+	 */
+	raft->leader_last_seen = raft_ev_monotonic_now(raft_loop());
+	raft->leader = 0;
+	bool was_seen = bit_test(&raft->leader_witness_map, raft->self);
+	assert(!was_seen);
+	(void)was_seen;
```

Comment:
This might also work:
```Diff
@@ -426,9 +426,7 @@ raft_leader_resign(struct raft *raft)
 	 */
 	raft->leader_last_seen = raft_ev_monotonic_now(raft_loop());
 	raft->leader = 0;
-	bool was_seen = bit_test(&raft->leader_witness_map, raft->self);
-	assert(!was_seen);
-	(void)was_seen;
+	assert(!bit_test(&raft->leader_witness_map, raft->self));
 	raft_schedule_broadcast(raft);
 }
```

Reply:
Yes, indeed.